Build failure: pipx, missing "test_skip_maintenance" from disabledTests

### Steps To Reproduce

Attempt to build pipx on nixos unstable

### Build log

```
...
Running phase: pytestCheckPhase
Executing pytestCheckPhase
============================= test session starts ==============================
platform linux -- Python 3.11.8, pytest-8.0.2, pluggy-1.4.0
rootdir: /build/source
configfile: pyproject.toml
collected 272 items / 176 deselected / 96 selected

tests/test_animate.py ...........                                        [ 11%]
tests/test_completions.py .                                              [ 12%]
tests/test_emojis.py ........................                            [ 37%]
tests/test_environment.py ..                                             [ 39%]
tests/test_interpreter.py ssssss...........                              [ 57%]
tests/test_list.py .F                                                    [ 59%]
tests/test_main.py ....                                                  [ 63%]
tests/test_package_specifier.py ...........................              [ 91%]
tests/test_pipx_metadata_file.py ....                                    [ 95%]
tests/test_shared_libs.py ..                                             [ 97%]
tests/test_util.py ..                                                    [100%]

=================================== FAILURES ===================================
____________________________ test_skip_maintenance _____________________________

pipx_temp_env = None

def test_skip_maintenance(pipx_temp_env):
>       assert not run_pipx_cli(["install", PKG["pycowsay"]["spec"]])
E       AssertionError: assert not 1
E        +  where 1 = run_pipx_cli(['install', 'pycowsay==0.0.0.2'])

tests/test_list.py:136: AssertionError
----------------------------- Captured stdout call -----------------------------

----------------------------- Captured stderr call -----------------------------
creating virtual environment...
creating shared libraries...
upgrading shared libraries...
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x
7ffff5533b50>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/pip/
...
```

### Additional context

Seems newer version of pipx has this new test "test_skip_maintenance" that attempts to pip install a package, which fails due to the network-sandboxed build environment. This causes the build to fail altogether. Hence, "test_skip_maintenance" should be added to the list `disabledTests` in https://github.com/NixOS/nixpkgs/blob/c1a17238bc1b6ba15ca7855b8125499683e5ab36/pkgs/development/python-modules/pipx/default.nix#L60.

### Notify maintainers

@yshym

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
- system: `"x86_64-linux"`
- host os: `Linux 6.6.22, NixOS, 24.05 (Uakari), 24.05pre601756.44d0940ea560`
- multi-user?: `yes`
- sandbox: `yes`
- version: `nix-env (Nix) 2.18.2`
- channels(root): `"nixos"`
- nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/pipx/default.nix
+++ b/pkgs/development/python-modules/pipx/default.nix
@@ -4,12 +4,11 @@
 , fetchFromGitHub
 , hatchling
 , hatch-vcs
-, importlib-metadata
 , packaging
-, pip
 , platformdirs
 , pytestCheckHook
 , pythonOlder
+, tomli
 , userpath
 , git
 }:
@@ -17,29 +16,29 @@
 buildPythonPackage rec {
   pname = "pipx";
   version = "1.4.3";
-  format = "pyproject";
+  pyproject = true;
 
-  disabled = pythonOlder "3.6";
+  disabled = pythonOlder "3.8";
 
   src = fetchFromGitHub {
-    owner = "pipxproject";
-    repo = pname;
+    owner = "pypa";
+    repo = "pipx";
     rev = "refs/tags/${version}";
     hash = "sha256-NxXOeVXwBhGqi4DUABV8UV+cDER0ROBFdgiyYTzdvuo=";
   };
 
-  nativeBuildInputs = [
+  build-system = [
     hatchling
     hatch-vcs
   ];
 
-  propagatedBuildInputs = [
+  dependencies = [
     argcomplete
     packaging
     platformdirs
     userpath
-  ] ++ lib.optionals (pythonOlder "3.8") [
-    importlib-metadata
+  ] ++ lib.optionals (pythonOlder "3.11") [
+    tomli
   ];
 
   nativeCheckInputs = [
@@ -83,7 +82,7 @@ buildPythonPackage rec {
   meta = with lib; {
     description = "Install and run Python applications in isolated environments";
     mainProgram = "pipx";
-    homepage = "https://github.com/pipxproject/pipx";
+    homepage = "https://github.com/pypa/pipx";
     changelog = "https://github.com/pypa/pipx/blob/${version}/CHANGELOG.md";
     license = licenses.mit;
     maintainers = with maintainers; [ yshym ];

--- a/pkgs/development/python-modules/pipx/default.nix
+++ b/pkgs/development/python-modules/pipx/default.nix
@@ -77,6 +77,7 @@ buildPythonPackage rec {
     "determination"
     "json"
     "test_list_short"
+    "test_skip_maintenance"
   ];
 
   meta = with lib; {

